clio-client/src/main/scala/org/broadinstitute/clio/client/commands/MoveWgsUbamCommand.scala

```diff
@@ -105,23 +137,45 @@ object MoveWgsUbamCommand
     ioUtil.deleteGoogleObject(path.get) match {
       case 0 => Future.successful(())
       case _ =>
-        Future.failed(new Exception("Delete files in google cloud failed"))
+        logger.error(
+          s"Deleting file in google cloud failed for path '${path.getOrElse("")}'"
+        )
+        Future.failed(new Exception())
     }
   }
 
   private def upsertUpdatedWgsUbam(webClient: ClioWebClient,
                                    config: BaseArgs): Future[HttpResponse] = {
-    val addResponse = webClient.addWgsUbam(
-      bearerToken = config.bearerToken.getOrElse(""),
-      input = TransferWgsUbamV1Key(
-        flowcellBarcode = config.flowcell.get,
-        lane = config.lane.get,
-        libraryName = config.libraryName.get,
-        location = Location.GCP
-      ),
-      transferWgsUbamV1Metadata =
-        TransferWgsUbamV1Metadata(ubamPath = config.ubamPath)
-    )
-    webClient.ensureOkResponse(addResponse)
+    implicit val executionContext: ExecutionContext = webClient.executionContext
+    webClient
+      .addWgsUbam(
+        bearerToken = config.bearerToken.getOrElse(""),
+        input = TransferWgsUbamV1Key(
+          flowcellBarcode = config.flowcell.get,
+          lane = config.lane.get,
+          libraryName = config.libraryName.get,
+          location = Location.GCP
+        ),
+        transferWgsUbamV1Metadata =
+          TransferWgsUbamV1Metadata(ubamPath = config.ubamPath)
+      )
+      .map(ensureOkResponse)
+  }
+
+  private def prettyKey(config: BaseArgs): String = {
```

Comment:
It seems like this should be a method on `ModelWgsUbamKey`, instead of a utility method here.

Reply:
I'm not sure that buys much. Here, I'm printing out the config arguments, not a ModelWgsUbamKey. The config might have empty fields, which ModelWgsUbamKey can't handle, since its parameters aren't defined as Option

Reply:
OK. Yeah, you'd have to first make a key here and then ask it to print, and I guess that would require some work.